analyzer/block/block.go

```diff
@@ -425,13 +466,20 @@ func NewAnalyzer(
 	if batchSize == 0 {
 		batchSize = defaultBatchSize
 	}
-	return &blockBasedAnalyzer{
+	a := &blockBasedAnalyzer{
 		blockRange:   blockRange,
 		batchSize:    batchSize,
 		analyzerName: name,
 		processor:    processor,
 		target:       target,
 		logger:       logger.With("analyzer", name, "mode", mode),
-		slowSync:     mode == analyzer.SlowSyncMode,
-	}, nil
+		queueLengthMetric: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: fmt.Sprintf("%s_queue_length", name),
+			Help: fmt.Sprintf("number of blocks left to process for the %s analyzer", name),
+		}),
+		slowSync: mode == analyzer.SlowSyncMode,
+	}
+	prometheus.MustRegister(a.queueLengthMetric)
```

Comment:
I think this might fail when you have parallelism, because prometheus only lets you register a metric once. This won't be a problem if you go with the suggested mini refactor.